airbyte-cdk/python/airbyte_cdk/logger.py

```diff
@@ -56,11 +56,26 @@ class AirbyteLogFormatter(logging.Formatter):
 
     def format(self, record: logging.LogRecord) -> str:
         """Return a JSON representation of the log message"""
-        message = super().format(record)
         airbyte_level = self.level_mapping.get(record.levelno, "INFO")
-        message = filter_secrets(message)
-        log_message = AirbyteMessage(type="LOG", log=AirbyteLogMessage(level=airbyte_level, message=message))
-        return log_message.json(exclude_unset=True)
+        if airbyte_level == "DEBUG":
+            extras = self.extract_extra_args_from_record(record)
+            debug_dict = {"type": "DEBUG", "message": record.getMessage(), "data": extras}
+            return filter_secrets(json.dumps(debug_dict))
```

Comment:
An alternative is to create an entirely new `AirbyteDebugMessage` that supplies the same flexibility to log variables. I didn't really envision the `--debug` command ever being invoked at prod level, but if it is, its probably worth investing in a new type of message into the protocol

Reply:
does this produce messages that are accepted by the platform? While we probably won't use --debug in production, I would imagine us using it on a dev machine or locally so we should make sure we can still run syncs with `--debug`

100% fine if the platform just ends up ignoring them

Reply:
https://github.com/airbytehq/airbyte/blob/master/docs/10-understanding-airbyte/02-airbyte-protocol.md#airbytelogmessage

Based on this it should treat any non message JSON as a defacto log message, might lose some prettification, but also saves us from committing Debug messages to being a first class citizen in the protocol right at this moment.